benchmarks/solitary_wave/solitary_wave.cc

```diff
@@ -381,10 +381,17 @@ namespace aspect
 
         virtual double reference_darcy_coefficient () const
         {
+          // Make sure we keep track of the initial composition manager and
+          // that it continues to live beyond the time when the simulator
+          // class releases its pointer to it.
+          if (initial_composition_manager == nullptr)
```

Comment:
Shouldn't this be in a initialize () function?

Reply:
This ended up not working because it is in the material model for which we call `initialize()` *before* the initial conditions objects are created (line 298 of `core.cc` for the `initialize()` call of the material model vs lines 308 and following for the initial condition creation). 

This was part of the reason for #4906 . I can address this as a follow-up to #4906 if you'd like me to, but would like to leave it as-is in this patch for now.

Reply:
Got it.